File: src/main/java/org/jukeboxmc/event/player/PlayerQuitEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerQuitEvent extends PlayerEvent {

    private String quitMessage;

    /**
     * Creates a new {@link PlayerQuitEvent}
     *
     * @param player      who has left the server
     * @param quitMessage which will be sent when the player leaves the server
     */
    public PlayerQuitEvent( Player player, String quitMessage ) {
        super( player );

        this.quitMessage = quitMessage;
    }

    /**
     * Retrieves the leave message
     *
     * @return a fresh {@link String}
     */
    public String getQuitMessage() {
        return this.quitMessage;
    }

    /**
     * Modifies the leave message
     *
     * @param quitMessage which should be modified
     */
    public void setQuitMessage( String quitMessage ) {
        this.quitMessage = quitMessage;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerQuitEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.player.Player

class PlayerQuitEvent(
    player: Player,
    private var quitMessage: String? = null
) : PlayerEvent(player) {

    fun getQuitMessage(): String? {
        return this.quitMessage
    }

    fun setQuitMessage(quitMessage: String?) {
        this.quitMessage = quitMessage
    }

}